types.go

```diff
@@ -24,6 +24,108 @@ func (t Timestamp) Parse() (time.Time, error) {
 	return time.Parse(time.RFC3339, string(t))
 }
 
+// NullString is a custom String type to allow for optional modifying strings that can be reset
+// the JSON Marshalling encoding is unique:
+// if the string is empty, it will be encoded as null
+// if the string is not empty, it will be encoded as a string
+type NullString string
+
+// MarshalJSON returns the JSON encoding of the NullString
+func (s NullString) MarshalJSON() ([]byte, error) {
+	if s == "" {
+		return json.Marshal(nil)
+	}
+
+	return json.Marshal(string(s))
+}
+
+func nullString(v interface{}) *NullString {
+	switch v.(type) {
+	case string:
+		if v.(string) == "" {
+			return nil
+		}
+
+		ns := NullString(v.(string))
+		return &ns
+	case NullString:
+		ns := NullString(v.(NullString))
+		return &ns
+	case *NullString:
+		return v.(*NullString)
+	default:
+		return nil
+	}
+}
```

Comment:
`switch` for type can create a variable of the case type automatically: https://tour.golang.org/methods/16

```suggestion
func nullString(i interface{}) *NullString {
	switch v := i.(type) {
	case string:
		if v == "" {
			return nil
		}

		ns := NullString(v)
		return &ns
	case NullString:
		return &v
	case *NullString:
		return v
	default:
		return nil
	}
}
```